pretyping/structures.mli

```diff
@@ -0,0 +1,163 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+
+(** A structure S is a non recursive inductive type with a single
+   constructor *)
+module Structure : sig
+
+(** A projection to a structure field *)
+type projection = {
+  proj_name : Names.Name.t;            (** field name *)
+  proj_true : bool;                    (** false for primitive records *)
```

Comment:
This is a weird field name.

Reply:
give me a better name...

Reply:
I'd rather call it `proj_prim` and negate the boolean.

Reply:
I think the right thing to do is to remove the field: if the record is primitive (boolean) then all projections are primitive as well.
But not this time...

Reply:
OK.